### PR TITLE
Allow the use of date_from and date_to in expressions

### PR DESCRIPTION
**Pull request by Martronic-SA**

### CODE REVIEW

**Comment by sbidoul** [comment]
Can you explain your use case?

**Comment by Martronic-SA** [comment]
For example I've made a EVA calculation for each month of 2016 which includes the Capital Charge on Operating Capital. This charge has to be reported equally on each month and the sum  of the months is the percentage x capital.
Sample here with a capital of 1 million and a percentage of 6%
`datetime.datetime.strptime(date_from,` '%Y-%m-%d').month*((0.06*1000000)/12)
total = 60000 and for each month: 5000, 10000, 15000, 20000, 25000, 30000, 35000, 40000, 45000, 50000, 55000, 60000

File: mis_builder/models/mis_safe_eval.py [hunk 1]
```diff
@@ -22,7 +23,10 @@ def mis_safe_eval(expr, locals_dict):
     """
     try:
         c = test_expr(expr, _SAFE_OPCODES, mode='eval')
-        globals_dict = {'__builtins__': _BUILTINS}
+        builtins = _BUILTINS.copy()
+        builtins.update({
+            'datetime':datetime})
+        globals_dict = {'__builtins__': builtins}
```
Comment (sbidoul) [inline mis_builder/models/mis_safe_eval.py]: It should not be necessary to fiddle with this low level code. Would it work to add this in prepare_locals_dict in mis_report?

**Comment by sbidoul** [comment]
@Martronic-SA Thanks for your contributions and patience.

I'm now back working on this and I'm planning to do a release with as many new features as possible before end of september.

I'll now flesh out the tests a little bit and merge at least some parts of this PR.

Could you rebase it on https://github.com/OCA/account-financial-reporting/pull/275 ?

**Comment by sbidoul** [comment]
This is now implemented in version 3 in a slightly different way: https://github.com/OCA/mis-builder/commit/802c24aa81af179c1f0d0245632803a18dae30a0

Thanks!

**Comment by Martronic-SA** [comment]
Great! Thank you!

**Pull request closed without merging**